mlir/test/lib/Dialect/SCF/TestLoopParametricTiling.cpp

```diff
@@ -40,6 +40,11 @@ class SimpleParametricLoopTilingPass
   }
 
   void runOnOperation() override {
+    if (sizes.empty()) {
+      getOperation()->emitError("expect non-empty outer loop sizes");
```

Comment:
I wouldn't use `getOperation()` because we don't need to attach any IR-related context.

I would also make the error message more clear on the fact that the error is about pass-options.

```suggestion
      emitError(UnknownLoc::get(getContext()), "missing `" + getArgument() + "` pass-option for outer loop sizes");
```

Reply:
@joker-eph 
When I use the unknown location here, the `expected-error` does not work as expected. The `verify-diagnostics` always prints the error message even if it matches. Is that the limitation of the error verification? Is there anything we can do as a workaround for that?

```
/Users/sasaki/dev/llvm-project/build/bin/mlir-opt -test-extract-fixed-outer-loops /Users/sasaki/dev/llvm-project/mlir/test/Transforms/invalid-outer-loop-size.mlir -verify-diagnostics
<unknown>:0: error: unexpected error: missing `test-outer-loop-sizes` pass-option for outer loop sizes
```